pkg/controllers/openshift-oauth-server.go

```diff
@@ -114,9 +115,12 @@ func (s *OpenShiftOAuth) Run(ctx context.Context, ready chan<- struct{}, stopped
 		close(ready)
 	}()
 
-	if err := oauth_apiserver.RunOAuthAPIServer(s.options, stopCh); err != nil {
-		return err
-	}
+	go func() {
+		if err := oauth_apiserver.RunOAuthAPIServer(s.options, stopCh); err != nil {
```

Comment:
Why create and pass in a new stop channel here, shouldn't this be `ctx.Done()`?

Reply:
Same reason, this api server does not receive any context anywhere,

Reply:
Sorry, I didn't fully understood your comment, moving to ctx.Done() which I remember doing and not working (channel directionality(, but it works.